packages/editor-ui/src/components/ValueSurvey.vue

```diff
@@ -1,3 +1,124 @@
+import { useI18n } from '@/composables/useI18n';
+import { ref, computed, watch } from 'vue';
+import { createEventBus } from 'n8n-design-system/utils';
+import { useTelemetry } from '@/composables/useTelemetry';
+import { useValueSurvey } from '@/stores/valueSurvey.store';
+
+const props = defineProps({
+	isActive: {
+		type: Boolean,
+	},
+});
+
+const rootStore = useRootStore();
+const i18n = useI18n();
+const toast = useToast();
+const telemetry = useTelemetry();
+
+const DEFAULT_TITLE = i18n.baseText('prompts.valueSurvey.recommendationQuestion');
+const GREAT_FEEDBACK_TITLE = i18n.baseText('prompts.valueSurvey.greatFeedbackTitle');
+const DEFAULT_FEEDBACK_TITLE = i18n.baseText('prompts.valueSurvey.defaultFeedbackTitle');
+const PRODUCT_TEAM_MESSAGE = i18n.baseText('prompts.productTeamMessage');
+const VERY_LIKELY_OPTION = i18n.baseText('prompts.valueSurvey.veryLikely');
+const NOT_LIKELY_OPTION = i18n.baseText('prompts.valueSurvey.notLikely');
+const SEND = i18n.baseText('prompts.valueSurvey.send');
+const YOUR_EMAIL_ADDRESS = i18n.baseText('prompts.valueSurvey.yourEmailAddress');
+
+const form = ref<{ value: string; email: string }>({ value: '', email: '' });
+const showButtons = ref(true);
+const modalBus = createEventBus();
+
+const modalTitle = computed(() => {
+	if (form?.value?.value !== '') {
+		if (Number(form.value) > 7) {
+			return GREAT_FEEDBACK_TITLE;
+		} else {
+			return DEFAULT_FEEDBACK_TITLE;
+		}
+	}
+
+	return DEFAULT_TITLE;
+});
+
+const isEmailValid = computed(
+	() => form?.value?.email && VALID_EMAIL_REGEX.test(String(form.value.email).toLowerCase()),
+);
+
+function closeDialog(): void {
+	if (form.value.value === '') {
+		telemetry.track('User responded value survey score', {
+			instance_id: rootStore.instanceId,
+			nps: '',
+		});
+
+		useValueSurvey().ignoreValueSurvey();
+	}
+	if (form.value.value !== '' && form.value.email === '') {
+		telemetry.track('User responded value survey email', {
+			instance_id: rootStore.instanceId,
+			email: '',
+			nps: form.value.value,
+		});
+	}
+}
+
+function onInputChange(value: string) {
+	form.value.email = value;
+}
+
+async function selectSurveyValue(value: string) {
+	form.value.value = value;
+	showButtons.value = false;
+
+	telemetry.track('User responded value survey score', {
+		instance_id: rootStore.instanceId,
+		nps: form.value.value,
+	});
+
+	useValueSurvey().respondValueSurvey();
+}
+
+async function send() {
+	if (isEmailValid.value) {
+		telemetry.track('User responded value survey email', {
+			instance_id: rootStore.instanceId,
+			email: form.value.email,
+			nps: form.value.value,
+		});
+
+		toast.showMessage({
+			title: i18n.baseText('prompts.valueSurvey.thanks'),
+			message: Number(form.value.value) >= 8 ? i18n.baseText('prompts.valueSurvey.reviewUs') : '',
+			type: 'success',
+			duration: 15000,
+		});
+
+		setTimeout(() => {
+			form.value.value = '';
+			form.value.email = '';
+			showButtons.value = true;
+		}, 1000);
```

Comment:
this modal has `keep-alive` set because of some technical issue that I don't remember now. So yes, it's necessary if the modal was to show in the same session, which was possible in the previous implementation or in testing. So will keep it.